### PR TITLE
source-spree-commerce contribution from Harmaton

### LINKED ISSUE #46632: [connector-request] Spree Commerce
This is a **suggestion** to create a connector for **Spree Commerce** using the UI Connector Builder.
The website for the connector is https://spreecommerce.org.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by Harmaton**
## What

This PR adds a new connector for source Spree Commerce (source-spree-commerce).

Contributor-provided connector description:

A coonecor for spree commerce Platform API
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `base_url` | `string` | Base_URL.  |  |
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| List Users |  | DefaultPaginator | ✅ |  ❌  |
| Variants |  | DefaultPaginator | ✅ |  ❌  |
| List Products |  | DefaultPaginator | ✅ |  ❌  |
| Payments |  | DefaultPaginator | ✅ |  ❌  |
| Payment Methods |  | DefaultPaginator | ✅ |  ❌  |
| List Orders |  | DefaultPaginator | ✅ |  ❌  |
| Menus |  | DefaultPaginator | ✅ |  ❌  |
| Addresses |  | DefaultPaginator | ✅ |  ❌  |
| Classifications |  | DefaultPaginator | ✅ |  ❌  |
| Adjustments |  | DefaultPaginator | ✅ |  ❌  |
| CMS Pages |  | DefaultPaginator | ✅ |  ❌  |
| CMS Sections |  | DefaultPaginator | ✅ |  ❌  |
| Shipments |  | DefaultPaginator | ✅ |  ❌  |
| Shipping Categories |  | DefaultPaginator | ✅ |  ❌  |
| States |  | DefaultPaginator | ✅ |  ❌  |
| Stock Items |  | DefaultPaginator | ✅ |  ❌  |
| Stock Locations |  | DefaultPaginator | ✅ |  ❌  |
| Stock Credit Categories |  | DefaultPaginator | ✅ |  ❌  |
| Store Credit Types |  | DefaultPaginator | ✅ |  ❌  |
| Store Credits |  | DefaultPaginator | ✅ |  ❌  |
| Tax Categories |  | DefaultPaginator | ✅ |  ❌  |
| Tax Rates |  | DefaultPaginator | ✅ |  ❌  |
| Wished Items |  | DefaultPaginator | ✅ |  ❌  |
| Wishlists |  | DefaultPaginator | ✅ |  ❌  |
| Zones |  | DefaultPaginator | ✅ |  ❌  |
| Webhook Subscribers |  | DefaultPaginator | ✅ |  ❌  |
| Taxonomies |  | DefaultPaginator | ✅ |  ❌  |
| Taxons |  | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

**Comment by natikgadzhi** [comment]
Huh, somehow the documentation md file wasn't generated ;( You could look at things like `docs/integrations/sources/basecamp.md` as an example and make one.

**Comment by Harmaton** [comment]
Is there a way to do this in the UI builder ?

**Comment by marcosmarxm** [comment]
Hello 👋 Due to the numerous contributions happening, I decided to create this [view](https://github.com/orgs/airbytehq/projects/104/views/3) to help you know when I plan to review your contribution. If your connector has some limited-time credential feel free to reach me out.

File: airbyte-integrations/connectors/source-spree-commerce/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,1094 @@
+version: 5.12.0
+
+type: DeclarativeSource
+
+description: A coonecor for spree commerce Platform API
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-spree-commerce/manifest.yaml:5]: Please provide a better description and check typos.

File: airbyte-integrations/connectors/source-spree-commerce/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,1094 @@
...
+    Tax Rates: true
+    Wished Items: true
+    Wishlists: true
+    Zones: true
+    Webhook Subscribers: true
+    Taxonomies: true
+    Taxons: true
+  testedStreams:
+    List Users:
+      streamHash: 81c15fe0f62280c1681ea8b538bacef3f52b115c
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-spree-commerce/manifest.yaml:897]: All streams weren't tested. I strongly suggest to build the connector using Airbyte Cloud.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
At the moment the connector wasn't tested and submited properly using the Connector Builder. Please check your submission, the review I made and create a new pull request. I'm closing this one.

**Comment by marcosmarxm** [comment]
If you're using Airbyte Open Source, please use latest version of the platform but I strongly recomend to use Airbyte Cloud to make contributions for the Hackathon.

**Pull request closed without merging**